### PR TITLE
Adding SVD type in the docstring of svds #19857

### PR DESCRIPTION
**Pull request by akaysh**
Added a detailed example too.

### CODE REVIEW

File: base/linalg/arnoldi.jl
Comment (andreasnoack) [inline base/linalg/arnoldi.jl]: > The algorithm produces `Vt`

This should be something like "internally `Vt` is stored and hence...", because in contrast to LAPACK, which produces `Vt`, the algorithm used by ARPACK actually produces `V`.

**Review by andreasnoack — CHANGES_REQUESTED** [review]
Great. Thanks for the PR.

File: base/linalg/arnoldi.jl
Comment (fredrikekre) [inline base/linalg/arnoldi.jl]: Can not use `sprand` in a doctest. Also, small floating numbers like this might not be the same all the time, see https://github.com/JuliaLang/julia/pull/20493 where this doctests originates from.

File: base/linalg/arnoldi.jl
Comment (fredrikekre) [inline base/linalg/arnoldi.jl]: remove these 3 empty lines?

File: base/linalg/arnoldi.jl
Comment (fredrikekre) [inline base/linalg/arnoldi.jl]: remove one empty line here, and below

File: base/linalg/arnoldi.jl
Comment (akaysh) [inline base/linalg/arnoldi.jl]: Oh I was not aware of this fact. Thanks for the info :)

File: base/linalg/arnoldi.jl
Comment (akaysh) [inline base/linalg/arnoldi.jl]: Alright!

File: base/linalg/arnoldi.jl
Comment (kshyatt) [inline base/linalg/arnoldi.jl]: You can use if it you (visibly) set a seed right before hand but it's strongly discouraged.

Comment (akaysh) [inline base/linalg/arnoldi.jl]: Okay I will use an int diagonal matrix.

File: base/linalg/arnoldi.jl
Comment (fredrikekre) [inline base/linalg/arnoldi.jl]: This example seem a bit superfluous when the exact same example is on the lines above.

File: base/linalg/arnoldi.jl
Comment (tkelman) [inline base/linalg/arnoldi.jl]: wrap long lines

File: base/linalg/arnoldi.jl
Comment (akaysh) [inline base/linalg/arnoldi.jl]: @fredrikekre Can you suggest what type of example would be more appropriate to understand SVD object?

Comment (fredrikekre) [inline base/linalg/arnoldi.jl]: I was just pointing out that the example you added are the same as the existing one -- so you could remove the existing one. No need to show the same thing twice.

Comment (akaysh) [inline base/linalg/arnoldi.jl]: Oh! okay.

File: base/linalg/arnoldi.jl
Comment (tkelman) [inline base/linalg/arnoldi.jl]: code highlight these individually, the and should not be highlighted

File: base/linalg/arnoldi.jl
Comment (StefanKarpinski) [inline base/linalg/arnoldi.jl]: This sentence is confusing – it seems to imply that we're using ARPACK rather than LAPACK, which therefore appears to contradict the claimed preference for `Vt`.

File: base/linalg/arnoldi.jl
Comment (StefanKarpinski) [inline base/linalg/arnoldi.jl]: You should only show the official way to access this, which is `Z[:S]` IIUC.

File: base/linalg/arnoldi.jl
Comment (StefanKarpinski) [inline base/linalg/arnoldi.jl]: Ditto.

File: base/linalg/arnoldi.jl
Comment (andreasnoack) [inline base/linalg/arnoldi.jl]: I'd just leave out the the "...because in contrast..." part

Comment (StefanKarpinski) [inline base/linalg/arnoldi.jl]: That makes the sentence even more confusing since it says:

1. `Vt` is more efficient to extract than `V`
2. the algorithm used by ARPACK actually produces `V`

This seems to be claiming that we store `Vt` but the algorithm we use produces `V`. Which is it? Also, this could stand to clarify which library we use for which computations – dense versus sparse – this isn't general knowledge for every reader.

Comment (akaysh) [inline base/linalg/arnoldi.jl]: @StefanKarpinski @andreasnoack Do we use LAPACK or ARPACK here?  I will only put the part we use and what it produces. (The other part makes it confusing)

**Comment by StefanKarpinski** [comment]
In the spirit of our new policy on docs, I propose that we merge this and then edit after? @andreasnoack? Would you mind doing the honors of editing to fix and clarify?

**Comment by tkelman** [comment]
is editing the branch enabled?

**Comment by StefanKarpinski** [comment]
@akaysh: if you enable us to push changes to this branch, then we can edit it in place, which is easier.

**Comment by akaysh** [comment]
@StefanKarpinski @tkelman Allow edits from maintainers is checked on this pull request.

**Comment by fredrikekre** [comment]
Needs a rebase, fixes #19857

**Review by andreasnoack — APPROVED** [review]

File: base/linalg/arnoldi.jl
Comment (Sacha0) [inline base/linalg/arnoldi.jl]: Should `s` still appear in the first line of the docstring, despite its removal from this part?

File: base/linalg/arnoldi.jl
Comment (tkelman) [inline base/linalg/arnoldi.jl]: does this render okay or does it need these lines to be indented with the rest of the bullet?

**Comment by andreasnoack** [comment]
This is stale. @akaysh If you want to finish this one, it is probably easier to start a new PR and copy over your changes since the relevant files are now in `stdlib`.

**Pull request closed without merging**
